src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 5;
+    public static final int FIGURE_SIDE = 50;
+    private Random random = new Random();
+
+    public int getRandomNumber() {
+        return random.nextInt(FIGURE_SIDE);
+    }
+
+    public Figure getRandomFigure() {
+        int figureNumber = random.nextInt(FIGURE_COUNT);
+        ColorSupplier supplier = new ColorSupplier();

```

Comment:
make color supplier a class variable as well. Like you did with Random object. We don't need to create each object per method call

Reply:
done